index.dd

```diff
@@ -234,19 +234,15 @@ void main()
 {
     import std.stdio, std.string, std.algorithm, std.conv;
 
-    // Reduce the RPN expression using a stack
-    readln.split.fold!((stack, op)
+    // arr is real[] and sym is the current symbol
+    readln.split.fold!((arr, sym) 
     {
-        switch (op)
-        {
-            // Generate operator switch cases statically
-            static foreach (c; "+-*/")
-                case [c]:
-                    return stack[0 .. $ - 2] ~
-                        mixin("stack[$ - 2] " ~ c ~
-                            " stack[$ - 1]");
-            default: return stack ~ op.to!real;
-        }
+        static foreach (c; "+-*/")
+            if (sym == [c])
+                return mixin("arr[0 .. $-2] ~
+                    (arr[$-2] " ~ c ~ " arr[$-1])");
```

Comment:
I think the original expression was clearer as it restricted the mixin to only the necessary expression. That helps tell the difference between the array append and the string appends. It could use a comment though:
```suggestion
                // replace the last 2 elements with the binary op
                return arr[0 .. $-2] ~
                    mixin("arr[$-2] " ~ c ~ " arr[$-1]");
```

Reply:
Should there be some comment explaining/showing a link to mixin or is that overdoing it?